### PR TITLE
feat(docker): support build-docker with dynamically input multiple build-args and secrets

### PR DESCRIPTION
**Pull request by gopkri24**
**Description**

Allow build-docker composite action to configure following:

- whether the user wants to push an image to registry or just build an image
- whether the user wants to automatically load the single-platform build result to docker images
- support to dynamically input multiple build-args
- support to dynamically input multiple secrets

**Changes**

* feat(docker): support build-docker with push and load inputs

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

**Comment by gopkri24** [comment]
The changes are tested in [here](https://github.com/turo/image-proxy/pull/191)

File: build-docker/action.yaml [hunk 1]
```diff
@@ -96,7 +116,11 @@ runs:
           VERSION=${{ inputs.image-version }}
           REVISION=${{ inputs.image-version }}
           BRANCH=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.branch'] }}
-        push: true
+          AWS_ACCESS_KEY_ID=${{ inputs.aws-access-key-id }}
+          AWS_SECRET_ACCESS_KEY=${{ inputs.aws-secret-access-key }}
+          AWS_REGION=${{ inputs.aws-region }}
```
Comment (tagoro9) [inline build-docker/action.yaml:121]: Why are these files needed? Not mentioned in the PR description

Comment (tagoro9) [reply build-docker/action.yaml:121]: They should probably be secrets (at least the secret key)

Comment (gopkri24) [reply build-docker/action.yaml]: Updated the mechanism to dynamically input multiple build args and secrets

File: build-docker/action.yaml [hunk 2]
```diff
@@ -96,10 +112,13 @@ runs:
           VERSION=${{ inputs.image-version }}
           REVISION=${{ inputs.image-version }}
           BRANCH=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.branch'] }}
```
Comment (tagoro9) [inline build-docker/action.yaml:114]: I think you are missing adding the `build-args` input here

**Review by tagoro9 — COMMENTED** [review]
Nice! Just one missing usage of an input

File: build-docker/action.yaml (hunk 2 above)
Comment (gopkri24) [reply build-docker/action.yaml:114]: Not sure if the github diff makes it weird, but I have added it 🙈 

image

Comment (gopkri24) [reply build-docker/action.yaml:114]: Tested this indirectly through another action [here](https://github.com/turo/automatic-pricing-msvc/pull/1131/files)

Comment (tagoro9) [reply build-docker/action.yaml:114]: Oh It's been a long day I guess, sorry

**Review by tagoro9 — APPROVED** [review]
